deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/api/CustomParamInitializer/ElementWiseParamInitializer.java

```diff
@@ -0,0 +1,182 @@
+                (org.deeplearning4j.nn.conf.layers.FeedForwardLayer) conf.getLayer();
+        int nIn = layerConf.getNIn();
+
+        int nWeightParams = nIn ;
+        INDArray weightView = paramsView.get(NDArrayIndex.point(0), NDArrayIndex.interval(0, nWeightParams));
+        INDArray biasView = paramsView.get(NDArrayIndex.point(0),
+                NDArrayIndex.interval(nWeightParams, nWeightParams + nIn));
+
+
+        params.put(WEIGHT_KEY, createWeightMatrix(conf, weightView, initializeParams));
+        params.put(BIAS_KEY, createBias(conf, biasView, initializeParams));
+        conf.addVariable(WEIGHT_KEY);
+        conf.addVariable(BIAS_KEY);
+
+        return params;
+    }
+
+    /**
+     * Return a map of gradients (in their standard non-flattened representation), taken from the flattened (row vector) gradientView array.
+     * The idea is that operates in exactly the same way as the the paramsView does in
+     * thus the position in the view (and, the array orders) must match those of the parameters
+     *
+     * @param conf         Configuration
+     * @param gradientView The flattened gradients array, as a view of the larger array
+     * @return A map containing an array by parameter type, that is a view of the full network gradients array
+     */
+    @Override
+    public Map<String, INDArray> getGradientsFromFlattened(NeuralNetConfiguration conf, INDArray gradientView) {
+        org.deeplearning4j.nn.conf.layers.FeedForwardLayer layerConf =
+                (org.deeplearning4j.nn.conf.layers.FeedForwardLayer) conf.getLayer();
+        int nIn = layerConf.getNIn();
+        int nOut = layerConf.getNOut();
+        int nWeightParams = nIn ;
+
+        INDArray weightGradientView = gradientView.get(NDArrayIndex.point(0), NDArrayIndex.interval(0, nWeightParams));
+        INDArray biasView = gradientView.get(NDArrayIndex.point(0),
+                NDArrayIndex.interval(nWeightParams, nWeightParams + nOut)); //Already a row vector
+
+        Map<String, INDArray> out = new LinkedHashMap<>();
+        out.put(WEIGHT_KEY, weightGradientView);
+        out.put(BIAS_KEY, biasView);
+
+        return out;
+    }
+    
+    protected INDArray createWeightMatrix(int nIn, int nOut, WeightInit weightInit, Distribution dist,
+                                          INDArray weightParamView, boolean initializeParameters) {
+        int[] shape = new int[] {nIn};
+
+        if (initializeParameters) {
+            INDArray ret = ElementWiseParamInitializer.initWeights(nIn, //Fan in
+                    nOut, //Fan out
+                    shape, weightInit, dist, DEFAULT_WEIGHT_INIT_ORDER,weightParamView);
+            return ret;
+        } else {
+//            System.out.println(weightParamView.shapeInfoToString());
+            return weightParamView;
+        }
+    }
+
+    public static INDArray initWeights(double fanIn, double fanOut, int[] shape, WeightInit initScheme,
```

Comment:
Is there a reason why the code from WeightInitUtil has to be copied to here?

Reply:
yes, the  matrix  w for the element wise layer is little different with the other, this is just for initialising the w  in the right form, which is a 1-d vector and the length = inputFeatureSize, otherwise it will initialise a 2-d matrix nIn*nOut

Reply:
The ```int[] shape``` argument specifies the shape. You should be able to pass [1,featureSize] to the existing code, to get the correct shape weights.